tests/test_TFNetworkLayer.py

```diff
@@ -1135,7 +1135,8 @@ def test_SplitLayer_search():
   with make_scope():
     net = TFNetwork(config=config, search_flag=True, train_flag=False, eval_flag=True)
     net.construct_from_dict({
-      "encoder": {"class": "linear", "activation": "tanh", "n_out": 5},
+      "encoder_seq": {"class": "linear", "activation": "tanh", "n_out": 5},
+      "encoder": {"class": "reduce", "mode": "sum", "from": ["encoder_seq"], "axis": "T"},
```

Comment:
Wait, does this work now? What you put into "from" is available in the rec unit as "data:source", but you don't use that. Maybe it works by chance because the encoder and rec layer time dims now match and therefore "base:encoder" is fed per timestep into the loop?

Reply:
What do you mean? Here encoder has time completely removed (i.e. just [B,D]). And it is in the rec-layer just as-is.

(Btw, off topic here, but if you need to have more generic unroll logic in the RecLayer, I thought about implementing sth like an UnrollLayer. Which would check of course that the time axis match such that this is possible. And which would use tf.TensorArray internally.)

Reply:
Oops, sorry, totally misread the code :D Yes, should be fine.

Reply:
But even without the reduce, this should work. Then you would just have this extra time dim inside the rec layer. (It would not do automatic unrolling. I think this would be confusing. It should be explicit.)